alchemist/alchemist-incarnation-protelis/src/main/java/it/unibo/alchemist/model/implementations/actions/SendToNeighbor.java

```diff
@@ -50,41 +58,16 @@ public SendToNeighbor cloneAction(final Node<Object> n, final Reaction<Object> r
                     .map(a -> (RunProtelisProgram<?>) a)
                     .collect(Collectors.toList());
             if (possibleRefs.size() == 1) {
-                return new SendToNeighbor((ProtelisNode<?>) n, reaction, possibleRefs.get(0));
+                return new SendToNeighbor((ProtelisNode<?>) n, this.getReaction(), possibleRefs.get(0));
```

Comment:
This is broken, for sure. You must pass the provided reaction

Reply:
If you mean the `r` parameter, ok, but then I think it was broken before, too (the returned action was built with the `reaction` field)

Reply:
Ugh
yes i believe it was broken before as well